### PR TITLE
DOC: add examples to random number generator pages

### PR DESCRIPTION
**Pull request by lastephey**

### CODE REVIEW

**Review by mattip — COMMENTED** [review]
Looks good, pending CI confirmation that the docstrings run reliably. It reads much better now

File: doc/source/reference/random/index.rst [hunk 1]
```diff
@@ -97,6 +99,30 @@ is wrapped with a `Generator`.
...
+0.3461018504257666
+>>> type(rfloat)
+<class 'float'>
+ 
+Here we use `default_rng` to create an instance of `Generator` to generate 3 
+random integers between 0 (inclusive) and 10 (exclusive):
+    
+>>> import numpy as np
+>>> rng = np.random.default_rng()
+>>> rints = rng.integers(low=0,high=10,size=3)
```
[the remark this answers is not in the document]
Comment (lastephey) [reply doc/source/reference/random/index.rst:121]: fixed

File: numpy/random/_generator.pyx [hunk 2]
```diff
@@ -4349,6 +4349,61 @@ def default_rng(seed=None):
...
+    generate the same random numbers again:
+
+    >>> import numpy as np
+    >>> rng = np.random.default_rng(seed=42)
+    >>> arr1 = rng.random((3, 3))
+    >>> arr1
+    array([[0.77395605, 0.43887844, 0.85859792],
+           [0.69736803, 0.09417735, 0.97562235],
+           [0.7611397 , 0.78606431, 0.12811363]])
+    >>> exit()
```
[the remark this answers is not in the document]
Comment (lastephey) [reply numpy/random/_generator.pyx:4405]: removed

File: numpy/random/_generator.pyx [hunk 3]
```diff
@@ -4349,6 +4349,61 @@ def default_rng(seed=None):
     -----
     If ``seed`` is not a `BitGenerator` or a `Generator`, a new `BitGenerator`
     is instantiated. This function does not manage a default global instance.
+    
+    Examples
+    --------
+    ``default_rng`` is the reccomended constructor for the random number class
+    ``Generator``. Here are several kinds of ways we can construct a random
```
[the remark this answers is not in the document]
Comment (lastephey) [reply numpy/random/_generator.pyx:4356]: fixed

File: doc/source/reference/random/index.rst [hunk 4]
```diff
@@ -113,25 +139,38 @@ bit generator-provided stream and transforms them into more useful
...
+pass it to `Generator`:
+
+>>> from numpy.random import Generator, PCG64
+>>> rg = Generator(PCG64(12345))
+>>> rg
+Generator(PCG64) at 0x7FBB88A5FC50
+>>> print(rg.random)
+<built-in method random of numpy.random._generator.Generator object at 0x7fbb88a5fc50>
+
+Similarly to use the older `MT19937` algorithm, one can instantiate it
```
[the remark this answers is not in the document]
Comment (lastephey) [reply doc/source/reference/random/index.rst:165]: changed

File: doc/source/reference/random/index.rst [hunk 5]
```diff
@@ -113,25 +139,38 @@ bit generator-provided stream and transforms them into more useful
 distributions, e.g., simulated normal random values. This structure allows
 alternative bit generators to be used with little code duplication.
 
-The `Generator` is the user-facing object that is nearly identical to
-`RandomState`. The canonical method to initialize a generator passes a
+The `Generator` is the user-facing object that is nearly identical to the
+legacy `RandomState`. The canonical method to initialize a generator passes a
 `PCG64` bit generator as the sole argument.
```
[the remark this answers is not in the document]
Comment (lastephey) [reply doc/source/reference/random/index.rst:144]: updated with suggested changes

File: doc/source/reference/random/index.rst [hunk 6]
```diff
@@ -113,25 +139,38 @@ bit generator-provided stream and transforms them into more useful
...
 One can also instantiate `Generator` directly with a `BitGenerator` instance.
-To use the older `MT19937` algorithm, one can instantiate it directly
-and pass it to `Generator`.
-
-.. code-block:: python
 
-  from numpy.random import Generator, MT19937
-  rg = Generator(MT19937(12345))
-  rg.random()
+To use the default `PCG64` algorithm, one can instantiate it directly and
```
[the remark this answers is not in the document]
Comment (lastephey) [reply doc/source/reference/random/index.rst:155]: changed

File: doc/source/reference/random/index.rst [hunk 7]
```diff
@@ -113,25 +139,38 @@ bit generator-provided stream and transforms them into more useful
...
-  rg.random()
+To use the default `PCG64` algorithm, one can instantiate it directly and 
+pass it to `Generator`:
+
+>>> from numpy.random import Generator, PCG64
+>>> rg = Generator(PCG64(12345))
+>>> rg
+Generator(PCG64) at 0x7FBB88A5FC50
+>>> print(rg.random)
+<built-in method random of numpy.random._generator.Generator object at 0x7fbb88a5fc50>
```
[the remark this answers is not in the document]
Comment (lastephey) [reply doc/source/reference/random/index.rst:163]: removed the rg.random and print(rg) here for clarity

**Review by mattip — APPROVED** [review]
Nit: one typo. Otherwise looks good.

**Comment by lastephey** [comment]
Thanks for fixing @charris

**Comment by charris** [comment]
Thanks @lastephey .

**Pull request merged**